modules/core/src/main/java/org/apache/ignite/internal/processors/security/SecurityUtils.java

```diff
@@ -153,6 +154,74 @@ public static SecurityContext nodeSecurityContext(Marshaller marsh, ClassLoader
         }
     }
 
+    /**
+     * Runs passed {@code runnable} with the security context associated
+     * with passed {@code secSubjId} if security is enabled.
+     *
+     * @param secSubjId Security subject id.
+     * @param ctx Grid kernal context.
+     * @param r Runnable.
+     */
+    public static void withContextIfNeed(UUID secSubjId, GridKernalContext ctx, RunnableX r)
+        throws IgniteCheckedException {
+        IgniteSecurity security = ctx.security();
+
+        try {
+            if (security.enabled() && secSubjId != null) {
+                try (OperationSecurityContext s = security.withContext(secSubjId)) {
+                    r.runx();
+                }
+            }
+            else
+                r.runx();
+        }
+        catch (Exception e) {
+            if (e instanceof IgniteCheckedException)
+                throw (IgniteCheckedException)e;
+
+            throw new IgniteCheckedException(e);
```

Comment:
Why do we need to change the exception here?
Can't we just throw it?

Reply:
I'll use the run method that doesn't throw an exception.